### ISSUE TITLE
Strictmode for Vertices

### ISSUE THREAD

**Issue by micha-nerdlichter** [orientechnologies/orientdb#3534]
Hi,
I want to use strictmode for vertices in a graph db.
I do the following:
````sql
create class V1 extends V
create class V2 extends V
alter class V1 strictmode=true;
alter class V2 strictmode=true;

create vertex V1
create vertex V2

create edge from (select from V1) to (select from V2)
```
and I get the following error:  **"Found additional field 'in_'. It cannot be added because the schema class 'V2' is defined as STRICT".**
So I think, OK I must create the in_ and out_ properties in the schema. I do
```sql
create property V1.out_ linklist
create property V2.in_ linklist
````
I figured LINKLIST is the only type that works here. I try ```create edge from (select from V1) to (select from V2)``` again and it works. I even can create a second edge with ```create edge from (select from V1) to (select from V2)```

But as soon as I start deleting edges, the trouble begins. ```delete edge #10:0``` works fine. But when I try to delete the second edge with ```delete edge #10:1``` I get the error **"com.orientechnologies.orient.core.exception.OValidationException: The field 'V1.out_' has been declared as LINKLIST but an incompatible type is used. Value: [V2#16:0{in_:[1]} v5]".**

So the in_ and out_ LINKLIST can't be empty? This is strange. How can I use vertices in strictmode then? Or am I doing the whole thing entirely wrong?

**Comment by a-unite** [comment]
This the same question I had here: https://github.com/orientechnologies/orientdb/issues/3289

**Comment by lvca** [comment]
Tested with OrientDB 2.0.x and works. Which release are you using?

**Comment by micha-nerdlichter** [comment]
I used 2.1-SNAPSHOT d76e920270
Also tried the official 2.0.1 download, all the same behaviour. So you can create an edge between two vertices in strictmode without any in_ and out_properties and don't get the "Field cannot be added" error?

**Comment by lvca** [comment]
I did:

```
create class V1 extends V
create class V2 extends V
alter class V1 strictmode=true;
alter class V2 strictmode=true;

create vertex V1
create vertex V2

create property V1.out_ linklist
create property V2.in_ linklist

create edge from (select from V1) to (select from V2)

delete edge #10:0
delete edge #10:1
```

Could you retry these steps, or is there something I skipped?

**Comment by micha-nerdlichter** [comment]
Yes, you have to do
```
create edge from (select from V1) to (select from V2)
create edge from (select from V1) to (select from V2)
```
otherwise you have only one edge and ```#10:1``` doesn't exist . The problem occurs as soon as there are multiple edges, I start deleting and there are only 2 left.

**Comment by lvca** [comment]
Sorry, this seems still valid.

**Comment by lvca** [comment]
Found the problem: on edge removal the setting graph.isAutoScaleEdgeType() is not considered, and a collection is transformed on single value. Fixed in 2.1.3.
